app/__init__.py

```diff
@@ -35,6 +35,11 @@ def create_app(config_name):
     login_manager.login_view = 'main.render_login'
     main_blueprint.config = application.config.copy()
 
+    @application.before_request
+    def remove_trailing_slash():
+        if request.path.endswith('/'):
+            return redirect(request.path[:-1], code=301)
```

Comment:
We could add `'?{}'.format(request.query_string)` to preserve the query string. Alternatively, if we don't expect any existing links/bookmarks with trailing slash and don't really care about the redirect it looks like there's a way to make flask handle links without slash and with slash the same way without any redirects. http://librelist.com/browser/flask/2011/2/27/re-automatic-removal-of-trailing-slashes/#043b1a0b6e841ab8e7d38bd7374cbb58